controllers/scheduler.go

```diff
@@ -191,6 +170,12 @@ func (s *Scheduler) cleanupDanglingWorkplacements(work *platformv1alpha1.Work) e
 	return nil
 }
 
+// Reconciles a WorkloadGroup by scheduling it to a Destination via a Workplacement.
+// If the Work is for a Resource Request, only one Workplacement will be created. If
+// this Workplacement already exists, it will be updated.
+// If the Work is for a Promise Dependency, a Workplacement will be created for each
+// Destination that matches the WorkloadGroup's DestinationSelectors. If any of these
+// Workplacements already exist, they will be updated.
```

Comment:
would it be better to inline the comments at the point they happen and leave the top level comment as a general documentation? i fear this can quickly get out of date and never updated.

Reply:
For resource Works, do we ever expect more than one WorkPlacement to be created for a WorkloadGroup (i.e. `len(existingWorkplacements) > 1`)?

I don't think we do, but we still iterate over them [below](https://github.com/syntasso/kratix/pull/59/files#diff-c70d9200060da1bcf5c33119592a7d74e3dab7b19cabf769686f0791372d8d0cL205)

Would it be better to assert that we only have one WorkPlacement before trying to update it?

Reply:
yeah, a resource can be now be scheduled across multiple destinations, so it may happen

Reply:
Just recapping what we just talked about:
- multi-destination scheduling allows a single `Work` to split across multiple `Destination`s
- it does this by defining a set of `WorkloadGroups` in a `Work`, each of which is scheduled to a specific (or all) `Destination`s (via `DestinationSelectors`)
- the `Scheduler` creates a _single_ `WorkPlacement` for each `WorkloadGroup` in order to place it in the given `Destination` (i.e. we _do_ only expect max. 1 `WorkPlacement` to be returned above)
- for dependency `Work`s, each `WorkloadGroup` is scheduled to _all_ matching `Destination`s
- for resource `Work`s, each `WorkloadGroup` is scheduled to _only one_ matching `Destination`

Shout if I've not quite covered anything right! 😊